code/modules/admin/verbs/adminhelp.dm

```diff
@@ -782,3 +782,35 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 				break
 
 	return potential_hits
+
+/**
+ * Checks a given message to see if any of the words contain an active admin's ckey with an @ before it
+ *
+ * Returns nothing if no pings are found, otherwise returns an associative list with ckey -> client
+ * Also modifies msg to underline the pings, then stores them in the key ["!new_msg"] for returning
+ *
+ * Arguments:
+ * * msg - the message being scanned
+ */
+/proc/check_admin_pings(msg)
+	//explode the input msg into a list
+	var/list/msglist = splittext(msg, " ")
+	var/list/admin_keys_to_clients = list()
+	var/list/admins_to_ping = list()
+
+	for(var/client/iter_admin in GLOB.admins)
+		admin_keys_to_clients[iter_admin.ckey] = iter_admin
+
+	var/i = 0
+	for(var/word in msglist)
+		i++
+		if(word[1] != "@")
+			continue
+		var/ckey_check = lowertext(copytext(word, 2))
+		if(admin_keys_to_clients[ckey_check])
```

Comment:
You could also check `GLOB.directory`, which has a map of ckey -> client, and from there check their holder and stuff.